gateway/provider.go

```diff
@@ -0,0 +1,44 @@
+package gateway
+
+import (
+	"context"
+
+	"github.com/dontpanicdao/caigo/types"
+)
+
+type GatewayProvider struct {
+	Gateway
+}
+
+func NewProvider(opts ...Option) *GatewayProvider {
+	return &GatewayProvider{
+		*NewClient(opts...),
+	}
+}
+
+func (p *GatewayProvider) BlockByHash(ctx context.Context, hash string) (*types.Block, error) {
```

Comment:
It looks like this doesn't implement the Provider interface

Reply:
ah good catch!